src/dbnode/encoding/istream.go

```diff
@@ -27,9 +27,9 @@ import (
 )
 
 const (
-	// defaultReaderSize is the default bufio.Reader size, we can keep this
-	// small as we rarely need to peek more than a byte or two at a time.
-	defaultReaderSize = 16
+	// defaultReaderSize is the default bufio.Reader size. Use a reasonable
+	// size to avoid excessive function call overhead.
+	defaultReaderSize = 128
```

Comment:
Could we make this configurable perhaps?  This showed up in the heap profiles at one point and shrinking it helped us reduce memory pressure:
https://github.com/m3db/m3/commit/f7d173ca5b37b7ce0ef9cbf86776a142cba648e0